### PR TITLE
Refactored installation process

### LINKED ISSUE #5: refactore the installation process
# Current installation process:
* Before installing the module, 
a. one must run ```python ../invenio-rdm-pure/invenio_rdm_pure/initial_setup.py```. in order to add some values for needed configurations: such as ```pure_username```, ```pure_api_key``` and so on... [1].
These values and variables then are saved to ```invenio_rdm_records/data_setup```.
b. creates a default user based on the giving values.

# Expected refactor
The installation process should put the module in a workable state.
- [x] move all these configuration variables from ```initial_setup.py```, to config variable of ```config.py```. that we the user can define these variables and their values also outside of the module.
- [x] ```./run-tests.sh``` should pass without errors in the module.
- [x] Create owner user for imported records with administrator rights at the installation process or at the first execution.
- [ ] @utnapischtim Add here what else .. there is to installing the module.

[1] https://github.com/tu-graz-library/invenio-rdm-pure/blob/0888fa020f233c6b9abbf3e14b51c746fd12a4e6/invenio_rdm_pure/initial_setup.py#L16

### PR DESCRIPTION
**Pull request by bengaal**
Refactored installation process. This PR closes TODO #3 of https://github.com/tu-graz-library/invenio-rdm-pure/issues/5, with which the PR closes https://github.com/tu-graz-library/invenio-rdm-pure/issues/5.
The changes include:
- Added pure user creation to function `get_pure_userid`. 
- Removed file `initial_setup.py`, because all the functionalities have been migrated.

### CODE REVIEW

File: invenio_rdm_pure/source/rdm/database.py [hunk 1]
```diff
@@ -59,17 +61,24 @@ def select_query(self, fields: str, table: str, filters={}):
...
-            return False
-        return response[0][0]
+    def get_pure_user_id(self):
+        """Gets the userId of the Pure user.
+
+        In case the user doesn't exist yet,
+        it is created with credentials defined in config.py.
+        """
+        datastore = current_app.extensions["security"].datastore
+        if datastore is not None:
```
Comment (mb-wali) [inline invenio_rdm_pure/source/rdm/database.py:71]: Add a short comment on ```datastore```. 
for example:

"""fetching user object with ```flask_security.datastore```."""

Question: wouldn't it be better to fetch with the class itself:

```
from flask_security.datastore import UserDatastore
invenio_pure_user = UserDatastore.get_user(invenio_pure_user_email)
```
and get rid of the datastore?
It's only better when writing tests for the module. and then we don't need to add the security to the extension.

**Review by mb-wali — COMMENTED** [review]
please write a short description of why these changes have been made. in the PR description level.
as I can tell it might be this part of #5.
> Create owner user for imported records with administrator rights at the installation process or at the first execution.

File: invenio_rdm_pure/source/rdm/database.py (hunk 1 above)
Comment (bengaal) [reply invenio_rdm_pure/source/rdm/database.py:71]: It sounds like a good idea, but I am not completely sure if that has the context of the module. 
I saw the user creation happening in this way [here](https://github.com/fair-data-austria/invenio-madmp/blob/74372ee794f81666f5e9cf08ef448c21b2e428be/tests/conftest.py#L139).

Additionally, as of now the implementation of get_user is missing:
```
def get_user(self, id_or_email):
        """Returns a user matching the specified ID or email address."""
        raise NotImplementedError
```

**Review by mb-wali — APPROVED** [review]
For now, seems ok, we can merge this- only in future, we want to have a look at creating the user  -
instead of using ```current_app.extensions["security"]``` which is extension from flask_security we should use ```invenio-accounts``` User model to create a new user.

**Pull request merged**